manager/allocator/network.go

```diff
@@ -1058,25 +1058,24 @@ func (a *Allocator) procTasksNetwork(ctx context.Context, onRetry bool) {
 		return
 	}
 
-	committed, err := a.store.Batch(func(batch *store.Batch) error {
+	err := a.store.Batch(func(batch *store.Batch) error {
 		for _, t := range allocatedTasks {
 			err := a.commitAllocatedTask(ctx, batch, t)
-
 			if err != nil {
 				log.G(ctx).WithError(err).Error("task allocation commit failure")
 				continue
 			}
+			delete(toAllocate, t.ID)
 		}
 
 		return nil
 	})
 
 	if err != nil {
 		log.G(ctx).WithError(err).Error("failed a store batch operation while processing tasks")
-	}
-
-	for _, t := range allocatedTasks[:committed] {
-		delete(toAllocate, t.ID)
+		for _, t := range allocatedTasks {
+			nc.unallocatedTasks[t.ID] = t
```

Comment:
I think this should be `toAllocate[t.ID] = t` given it may point to `nc.unallocatedTasks` or to `nc.pendingTasks` based on the ctx this function is called within.